SAP2000_Adapter/CRUD/Read/SurfaceProperty.cs

```diff
@@ -61,42 +61,35 @@ private List<ISurfaceProperty> ReadSurfaceProperty(List<string> ids = null)
-                    panelConstant.CustomData.Add("BendingThickness", bending);
-                    panelConstant.CustomData.Add("Color", color);
-                    panelConstant.CustomData.Add("Notes", notes);
-                    panelConstant.CustomData.Add("GUID", guid);
-
-                    if (hasModifiers)
-                    {
-                        panelConstant.CustomData.Add("MembraneF11Modifier", modifiers[0]);
-                        panelConstant.CustomData.Add("MembraneF22Modifier", modifiers[1]);
-                        panelConstant.CustomData.Add("MembraneF12Modifier", modifiers[2]);
-                        panelConstant.CustomData.Add("BendingM11Modifier", modifiers[3]);
-                        panelConstant.CustomData.Add("BendingM22Modifier", modifiers[4]);
-                        panelConstant.CustomData.Add("BendingM12Modifier", modifiers[5]);
-                        panelConstant.CustomData.Add("ShearV13Modifier", modifiers[6]);
-                        panelConstant.CustomData.Add("ShearV23Modifier", modifiers[7]);
-                        panelConstant.CustomData.Add("MassModifier", modifiers[8]);
-                        panelConstant.CustomData.Add("WeightModifier", modifiers[9]);
-                    }
-
-                    propertyList.Add(panelConstant);
-                }
+                m_model.PropArea.GetModifiers(id, ref modifiers);
+                                
+                ConstantThickness panelConstant = new ConstantThickness();
+                panelConstant.CustomData[AdapterIdName] = id;
+                panelConstant.Name = id;
+                panelConstant.Material = bhomMaterials[materialName];
+                panelConstant.Thickness = thickness;
+                panelConstant.CustomData.Add("MaterialAngle", matAng);
+                panelConstant.CustomData.Add("BendingThickness", bending);
+                panelConstant.CustomData.Add("Color", color);
+                panelConstant.CustomData.Add("Notes", notes);
+                panelConstant.CustomData.Add("GUID", guid);
+                
+                panelConstant.CustomData.Add("MembraneF11Modifier", modifiers[0]);
+                panelConstant.CustomData.Add("MembraneF22Modifier", modifiers[1]);
+                panelConstant.CustomData.Add("MembraneF12Modifier", modifiers[2]);
+                panelConstant.CustomData.Add("BendingM11Modifier", modifiers[3]);
+                panelConstant.CustomData.Add("BendingM22Modifier", modifiers[4]);
+                panelConstant.CustomData.Add("BendingM12Modifier", modifiers[5]);
+                panelConstant.CustomData.Add("ShearV13Modifier", modifiers[6]);
+                panelConstant.CustomData.Add("ShearV23Modifier", modifiers[7]);
+                panelConstant.CustomData.Add("MassModifier", modifiers[8]);
+                panelConstant.CustomData.Add("WeightModifier", modifiers[9]);                
```

Comment:
We have a method called "ApplyModifiers" in Structure_Engine that is meant to be used for this.

Will formalise this into a fragment this milestone, is the plan at least.

Reply:
Ah, thanks! can I use it now or should I wait till fragment is released?

Reply:
I can use it now! Thanks @IsakNaslundBh

Reply:
answered yourself, but yet. Was planning to make those methods point to the fragment as soon as it has been added :)